src/assets/CollectiblesController.ts

```diff
@@ -236,14 +237,27 @@ export class CollectiblesController extends BaseController<
   ): Promise<CollectibleMetadata> {
     const tokenURI = this.getCollectibleApi(contractAddress, tokenId);
     let collectibleInformation: ApiCollectible;
-
     /* istanbul ignore if */
-    if (this.openSeaApiKey) {
-      collectibleInformation = await handleFetch(tokenURI, {
-        headers: { 'X-API-KEY': this.openSeaApiKey },
-      });
-    } else {
+    try {
       collectibleInformation = await handleFetch(tokenURI);
+    } catch (e) {
+      logOrRethrowError(e);
+
+      if (this.openSeaApiKey) {
+        collectibleInformation = await handleFetch(
+          `https://api.opensea.io/api/v1/asset/${contractAddress}/${tokenId}`,
+          {
+            headers: { 'X-API-KEY': this.openSeaApiKey },
+          },
+        );
+      } else {
+        return {
```

Comment:
Could this be an assignment to `collectibleInformation` instead? That would avoid having to encode the defaults in two places.

Reply:
> That would avoid having to encode the defaults in two places.

Not sure I follow this? Where else are the defaults being encoded within this function?

Reply:
Oh you mean the type is an encoding of the defaults

Reply:
If we want to set it to the default version of the expected type for collectibleInformation - `ApiCollectible` - it would involve duplicating the encoded defaults again: 
```
 collectibleInformation = {
        name: null,
        description: null,
        asset_contract: {
          address: '',
          asset_contract_type: null,
          created_date: null,
          schema_name: null,
          symbol: null,
          total_supply: null,
          description: null,
          external_link: null,
          collection: {
            name: null,
            image_url: null,
          }
        },
        token_id: '',
        num_sales: null,
        background_color: null,
        image_url: null,
        image_preview_url: null,
        image_thumbnail_url: null,
        image_original_url: null,
        animation_url: null,
        animation_original_url: null,
        external_link: null,
        creator: {
          user: { username: '' },
          profile_img_url: '',
          address: '',
        },
        last_sale: {
          event_timestamp: '',
          total_price: '',
          transaction: { transaction_hash: '', block_hash: '' },
        },
      };
```

Reply:
>Oh you mean the type is an encoding of the defaults

Sorry no, this is not what I meant. I don't understand the later comments in this thread either, maybe because of this misunderstanding.

All I'm saying is that the defaults are set in two places. Here, and below on lines 283-285. If this line is replaced with `collectibleInformation = {}`, it would be functionally equivalent but the defaults wouldn't need to be duplicated in two places.

Reply:
The problem is that `collectibleInformation` is type `ApiCollectible` (line 239). So it can't be set to an empty object because that type is:

```
ApiCollectible {
  token_id: string;
  num_sales: number | null;
  background_color: string | null;
  image_url: string | null;
  image_preview_url: string | null;
  image_thumbnail_url: string | null;
  image_original_url: string | null;
  animation_url: string | null;
  animation_original_url: string | null;
  name: string | null;
  description: string | null;
  external_link: string | null;
  asset_contract: ApiCollectibleContract;
  creator: ApiCollectibleCreator;
  last_sale: ApiCollectibleLastSale | null;
}
```

Reply:
`collectibleInformation` is type `ApiCollectible` while the return type for the function is `CollectibleMetadata`:

interface CollectibleMetadata {
  name: string | null;
  description: string | null;
  image: string | null;
  standard: string | null;
  favorite?: boolean;
  numberOfSales?: number;
  backgroundColor?: string;
  imagePreview?: string;
  imageThumbnail?: string;
  imageOriginal?: string;
  animation?: string;
  animationOriginal?: string;
  externalLink?: string;
  creator?: ApiCollectibleCreator;
  lastSale?: ApiCollectibleLastSale;
}

Reply:
Why is that a problem? Can't you update the type of that variable to `CollectibleMetadata`? It must be assignable to that, given that that is what is being returned. I don't follow why this difference in types is a problem.

Reply:
I might still be missing something about what you're suggesting, but the issue is that at the bottom of the function (after we've fetched data from either the proxy or directly from OpenSea) we take the shape of `ApiCollectible` (the expected return type from the API calls) and change it to the `CollectibleMetadata` shape which is how we want to store the data. In order re-type the `collectibleInformation` as `CollectibleMetadata` we would have to replace `handleFetch` with another function that combines the job of fetching and then (if successful) reshaping the data from `ApiCollectible` to `CollectibleMetadata` for each attempt.

Reply:
Ok, I see what you mean. I won't block on this but I still think this function is unnecessarily complex, and that it would be better to set default values in one place rather than two. But at least they're encoded in the return type as well I guess.

Reply:
I agree it could use a refactor! But perhaps we can leave that for another PR.